Settings tab: Buy/Sell Fees

fabriziovigevani opened the pull request:
fixes #9 

Includes Buy/Sell fees basic panel with the option to edit them and Side panel for editing.


**Preview**

![Screen Shot 2020-05-28 at 3 06 03 PM](https://user-images.githubusercontent.com/22663232/83177064-d4bf6980-a0f4-11ea-9caf-12580445e92d.png)

![Screen Shot 2020-05-28 at 2 34 55 PM](https://user-images.githubusercontent.com/22663232/83177095-e143c200-a0f4-11ea-9d93-b09081ca0fcd.png)

fabriziovigevani committed:
Settings: Add Fees panel

fabriziovigevani committed:
bn-utils: Add helper functions to adjust percentages

fabriziovigevani committed:
Fees: Implement Fee panel and Side panel for editing fees

fabriziovigevani committed:
Script.js: Fix lint

fabriziovigevani committed:
Edit fees Side panel: Rename title

willjgriff commented:
Shame about that whitespace in the collateralization ratios box, it used to be able to have multiple collaterals in it but now it only has one, would be better to just have 3 boxes in a line. Is there anything easy that can be done there?

I would consider adding an info panel to the top of side panel input with a heading something like "Action" and detail "Update the fee taken from all buy/sell orders that is deposited into the organisation's funding pool".

Also maybe "Update Fees" instead of "Edit fees"? Edit makes me think of Word. At least could we capitalise the F in "fees"?

fabriziovigevani commented:
>  Is there anything easy that can be done there?

Yeah true that is looks a little bit weird with all that space. Yeah we can align all boxes in one row for sure.

> I would consider adding an info panel to the top of side panel input with a heading something like "Action" and detail "Update the fee taken from all buy/sell orders that is deposited into the organisation's funding pool".

👍 

> Also maybe "Update Fees" instead of "Edit fees"? Edit makes me think of Word. At least could we capitalise the F in "fees"?

Fair point. Yeah we can change it to `Update`

fabriziovigevani committed:
Do suggestions done by Will
Rephrase Edit to Update
Add Info to UpdateFee side panel
Settings: Rearrange layout

fabriziovigevani commented:
@willjgriff Updated PR with suggestions and screenshots.

Regarding the layout, do you think it's fine like this or maybe we could reorder the boxes (e.g moving `collateralization ratios` to the end)?

fabriziovigevani commented on apps/aragon-fundraising/app/src/screens/Reserves.js:
```diff
@@ -61,79 +60,80 @@ export default () => {
+          </div>
+        </Box>
+
+        <div>
```
`Box` components direct siblings set a `margin top` by default. This enclosing div is to prevent Boxes to be direct siblings and prevent setting margins

willjgriff replied:
Maybe leave that as a comment in the code too.

willjgriff commented:
> Regarding the layout, do you think it's fine like this or maybe we could reorder the boxes (e.g moving `collateralization ratios` to the end)?

Mmm, it's hard innit, they couldn't have even widths could they? Or do the boxes have an unchangeable width? Otherwise it's fine as is or with collateralization ratio at the end, whatever looks best to you.
